sdk/eventhub/event-hubs/src/impl/partitionGate.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+export class PartitionGate {
```

Comment:
Can you explain what the `PartitionGate` is and what it's meant to do?

Reply:
It makes it so you can't (from the same consumer client) start up multiple consumers against the "same" partitions.

Reply:
Oh...perhaps you meant "can you comment your work?" :)

I've added in some comments.

Reply:
Thank you!